### PR TITLE
fix(node): wrong `worker_threads.terminate()` return value

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes https://github.com/denoland/deno/issues/23801

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Just checked and Node returns `0` from `.terminate()`.

**Comment by marvinhagemeister** [comment]
Not sure about the test failures. I'm unable to reproduce them locally.

File: tests/unit_node/worker_threads_test.ts [hunk 1]
```diff
@@ -516,6 +516,29 @@ Deno.test({
   },
 });
 
+Deno.test({
+  name: "[node/worker_threads] Returns terminate promise with exit code",
+  async fn() {
+    const deferred = Promise.withResolvers<void>();
+    const worker = new workerThreads.Worker(
+      `parentPort.postMessage("ok");`,
```
Comment (bartlomieju) [inline tests/unit_node/worker_threads_test.ts:524]: I believe you need to import it explicitly

Comment (bartlomieju) [inline tests/unit_node/worker_threads_test.ts:524]:
```suggestion
      `import { parentPort } from "node:worker_threads";
        parentPort.postMessage("ok");`,
```

File: ext/node/polyfills/worker_threads.ts [hunk 2]
```diff
@@ -281,6 +282,7 @@ class NodeWorker extends EventEmitter {
       op_host_terminate_worker(this.#id);
     }
     this.emit("exit", 1);
+    return PromiseResolve(0);
```
Comment (bartlomieju) [inline ext/node/polyfills/worker_threads.ts:285]: Shouldn't it return the same code as `exit`?

File: tests/unit_node/worker_threads_test.ts (hunk 1 above)
Comment (marvinhagemeister) [reply tests/unit_node/worker_threads_test.ts:524]: Ohh 💩🙈

File: ext/node/polyfills/worker_threads.ts (hunk 2 above)
Comment (marvinhagemeister) [reply ext/node/polyfills/worker_threads.ts:285]: Need to compare with nice. Haven't checked what they emit

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**